// @salesforce/design-system-react/module/components/story-based-tests.js
/*
 * STORY-BASED SNAPSHOT TESTING
 *
 * Please add stories below to use Storybook Stories (http://localhost:9001)
 * as the basis for DOM and image snapshots to allow markup and visual
 *  regression testing. All new components should be added below to enable
 * DOM and image snapshots. With enough Storybook story examples, these two
 * methods should be significant to fully test components with the exception
 * of callback testing. Please test callback props with the Mocha framework
 * (http://localhost:8001). `tests/story-based-tests.snapshot-test.js` looks
 * at this file and uses the following stories to create snapshots.
 *
 * This library is transitioning to story-based testing, but not all components
 * are able to be tested without a DOM. This file should eventually go away and
 * `storybook-stories.js` used for the basis of snapshot creation.
 */
import _Blank from '../tests/initial-blank-stories';
export { _Blank as Blank };
import _Accordion from '../components/accordion/__docs__/storybook-stories';
export { _Accordion as Accordion };
import _Alert from '../components/alert/__docs__/storybook-stories';
export { _Alert as Alert };
import _Avatar from '../components/avatar/__docs__/storybook-stories';
export { _Avatar as Avatar };
import _BrandBand from '../components/brand-band/__docs__/storybook-stories';
export { _BrandBand as BrandBand };
import _Breadcrumb from '../components/breadcrumb/__docs__/storybook-stories';
export { _Breadcrumb as Breadcrumb };
import _BuilderHeader from '../components/builder-header/__docs__/storybook-stories';
export { _BuilderHeader as BuilderHeader };
import _Button from '../components/button/__docs__/storybook-stories';
export { _Button as Button };
import _ButtonGroup from '../components/button-group/__docs__/storybook-stories';
export { _ButtonGroup as ButtonGroup };
import _ButtonStateful from '../components/button-stateful/__docs__/storybook-stories';
export { _ButtonStateful as ButtonStateful };
import _Card from '../components/card/__docs__/storybook-stories';
export { _Card as Card };
import _Carousel from '../components/carousel/__docs__/storybook-stories';
export { _Carousel as Carousel };
import _Checkbox from '../components/checkbox/__docs__/storybook-stories';
export { _Checkbox as Checkbox };
import _ColorPicker from '../components/color-picker/__docs__/storybook-stories';
export { _ColorPicker as ColorPicker };
import _Combobox from '../components/combobox/__docs__/storybook-stories';
export { _Combobox as Combobox };
import _ExpandableSection from '../components/expandable-section/__docs__/storybook-stories';
export { _ExpandableSection as ExpandableSection };
import _Filter from '../components/filter/__docs__/storybook-stories';
export { _Filter as Filter };
import _DataTable from '../components/data-table/__docs__/storybook-stories';
export { _DataTable as DataTable };
import _DynamicIcon from '../components/dynamic-icon/__docs__/storybook-stories';
export { _DynamicIcon as DynamicIcon }; // export Dropdown from '../components/menu-dropdown/__docs__/storybook-stories';

import _GlobalHeader from '../components/global-header/__docs__/storybook-stories';
export { _GlobalHeader as GlobalHeader };
import _Input from '../components/input/__docs__/storybook-stories';
export { _Input as Input };
import _Icon from '../components/icon/__docs__/storybook-stories';
export { _Icon as Icon };
import _Illustration from '../components/illustration/__docs__/storybook-stories';
export { _Illustration as Illustration };
import _MediaObject from '../components/media-object/__docs__/storybook-stories';
export { _MediaObject as MediaObject };
import _PageHeader from '../components/page-header/__docs__/storybook-stories';
export { _PageHeader as PageHeader }; // export Pill from '../components/pill/__docs__/storybook-stories';

import _PillContainer from '../components/pill-container/__docs__/storybook-stories';
export { _PillContainer as PillContainer };
import _Popover from '../components/popover/__docs__/storybook-stories';
export { _Popover as Popover };
import _ProgressIndicator from '../components/progress-indicator/__docs__/storybook-stories';
export { _ProgressIndicator as ProgressIndicator };
import _ProgressBar from '../components/progress-bar/__docs__/storybook-stories';
export { _ProgressBar as ProgressBar };
import _Spinner from '../components/spinner/__docs__/storybook-stories';
export { _Spinner as Spinner };
import _ProgressRing from '../components/progress-ring/__docs__/storybook-stories';
export { _ProgressRing as ProgressRing }; // export RadioGroup from '../components/radio-group/__docs__/storybook-stories';
// export RadioButtonGroup from '../components/radio-button-group/__docs__/storybook-stories';

import _ScopedNotification from '../components/scoped-notification/__docs__/storybook-stories';
export { _ScopedNotification as ScopedNotification }; // export Search from '../components/input/__docs__/search/storybook-stories';
// export Slider from '../components/slider/__docs__/storybook-stories';
// export SplitView from '../components/split-view/__docs__/storybook-stories';
// export Tabs from '../components/tabs/__docs__/storybook-stories';
// export Textarea from '../components/textarea/__docs__/storybook-stories';
// export Toast from '../components/toast/__docs__/storybook-stories';

import _Tooltip from '../components/tooltip/__docs__/storybook-stories';
export { _Tooltip as Tooltip };
import _Tree from '../components/tree/__docs__/storybook-stories';
export { _Tree as Tree }; // export VerticalNavigation from '../components/vertical-navigation/__docs__/storybook-stories';

/*
 * The following components are not compatible Jest testing because they
 * need a real DOM. These components should be updated to allow testing
 * with Jest. This is typically done by gaurding references to `document`.
 */
// export AppLauncher from '../components/app-launcher/__docs__/storybook-stories';
// export GlobalNavigationBar from '../components/global-navigation-bar/__docs__/storybook-stories';
// export DatePicker from '../components/date-picker/__docs__/storybook-stories';
// export IconSettings from '../components/icon-settings/__docs__/storybook-stories';
// export Modal from '../components/modal/__docs__/storybook-stories';
// export Panel from '../components/panel/__docs__/storybook-stories';
// export TimePicker from '../components/time-picker/__docs__/storybook-stories';
// DEPRECATED
// export Lookup from '../components/lookup/__docs__/storybook-stories';
// export Notification from '../components/notification/__docs__/storybook-stories';
// export Picklist from '../components/menu-picklist/__docs__/storybook-stories';
