// @salesforce/design-system-react/module/components/page-header/__examples__/object-home.js
function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import IconSettings from "../../../../components/icon-settings";
import PageHeader from "../../../../components/page-header"; // `~` is replaced with design-system-react at runtime

import Button from "../../../../components/button";
import ButtonGroup from "../../../../components/button-group";
import Dropdown from "../../../../components/menu-dropdown";
import DropdownTrigger from "../../../../components/menu-dropdown/button-trigger";

var Example =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Example, _React$Component);

  function Example() {
    _classCallCheck(this, Example);

    return _possibleConstructorReturn(this, _getPrototypeOf(Example).apply(this, arguments));
  }

  _createClass(Example, [{
    key: "render",
    value: function render() {
      var controls = function controls() {
        return React.createElement("div", null, React.createElement(ButtonGroup, null, React.createElement(Button, {
          label: "New Lead"
        }), React.createElement(Button, {
          label: "Import Leads"
        }), React.createElement(Dropdown, {
          align: "right",
          assistiveText: {
            icon: 'More Options'
          },
          iconCategory: "utility",
          iconName: "down",
          iconVariant: "border-filled",
          id: "page-header-dropdown-object-home-nav-right",
          options: [{
            label: 'Menu Item One',
            value: 'A0'
          }, {
            label: 'Menu Item Two',
            value: 'B0'
          }, {
            label: 'Menu Item Three',
            value: 'C0'
          }, {
            type: 'divider'
          }, {
            label: 'Menu Item Four',
            value: 'D0'
          }]
        })));
      };

      var actions = function actions() {
        return React.createElement("div", null, React.createElement(Dropdown, {
          align: "right",
          id: "page-header-dropdown-object-home-content-right",
          options: [{
            label: 'Menu Item One',
            value: 'A0'
          }, {
            label: 'Menu Item Two',
            value: 'B0'
          }, {
            label: 'Menu Item Three',
            value: 'C0'
          }, {
            type: 'divider'
          }, {
            label: 'Menu Item Four',
            value: 'D0'
          }]
        }, React.createElement(DropdownTrigger, null, React.createElement(Button, {
          assistiveText: {
            icon: 'List View Controls'
          },
          className: "slds-m-right_xx-small",
          iconCategory: "utility",
          iconName: "settings",
          iconVariant: "more"
        }))), React.createElement(Dropdown, {
          align: "right",
          assistiveText: {
            icon: 'Change view'
          },
          iconCategory: "utility",
          iconName: "settings",
          iconVariant: "more",
          id: "page-header-dropdown-object-home-content-right-2",
          options: [{
            label: 'Menu Item One',
            value: 'A0'
          }, {
            label: 'Menu Item Two',
            value: 'B0'
          }, {
            label: 'Menu Item Three',
            value: 'C0'
          }, {
            type: 'divider'
          }, {
            label: 'Menu Item Four',
            value: 'D0'
          }]
        }, React.createElement(DropdownTrigger, null, React.createElement(Button, {
          assistiveText: {
            icon: 'Change view'
          },
          className: "slds-m-right_xx-small",
          iconCategory: "utility",
          iconName: "table",
          iconVariant: "more",
          variant: "icon"
        }))), React.createElement(Button, {
          assistiveText: {
            icon: 'Edit List'
          },
          iconCategory: "utility",
          iconName: "edit",
          iconVariant: "border",
          variant: "icon"
        }), React.createElement(Button, {
          assistiveText: {
            icon: 'Refresh'
          },
          iconCategory: "utility",
          iconName: "refresh",
          iconVariant: "border",
          variant: "icon"
        }), React.createElement(ButtonGroup, null, React.createElement(Button, {
          assistiveText: {
            icon: 'Charts'
          },
          iconCategory: "utility",
          iconName: "chart",
          iconVariant: "border",
          variant: "icon"
        }), React.createElement(Button, {
          assistiveText: {
            icon: 'Filters'
          },
          iconCategory: "utility",
          iconName: "filterList",
          iconVariant: "border",
          variant: "icon"
        })));
      };

      return React.createElement(IconSettings, {
        iconPath: "/assets/icons"
      }, React.createElement(PageHeader, {
        onRenderActions: actions,
        iconAssistiveText: "User",
        iconCategory: "standard",
        iconName: "lead",
        info: "10 items \u2022 sorted by name",
        label: "Leads",
        onRenderControls: controls,
        title: React.createElement("h1", {
          className: "slds-page-header__title slds-p-right_x-small"
        }, React.createElement(Dropdown, {
          id: "page-header-dropdown-object-home-header",
          options: [{
            label: 'Menu Item One',
            value: 'A0'
          }, {
            label: 'Menu Item Two',
            value: 'B0'
          }, {
            label: 'Menu Item Three',
            value: 'C0'
          }, {
            type: 'divider'
          }, {
            label: 'Menu Item Four',
            value: 'D0'
          }]
        }, React.createElement(DropdownTrigger, null, React.createElement(Button, {
          className: "slds-button_reset slds-type-focus",
          iconCategory: "utility",
          iconName: "down",
          iconPosition: "right",
          label: "Dropdown",
          responsive: true,
          variant: "base"
        })))),
        truncate: true,
        variant: "object-home"
      }));
    }
  }]);

  return Example;
}(React.Component);

_defineProperty(Example, "displayName", 'ObjectHomePageHeaderExample');

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
