// @salesforce/design-system-react/module/components/page-header/private/base/index.js
/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */

/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
import React from 'react';
import PropTypes from 'prop-types';
import MediaObject from '../../../media-object';
var displayName = 'PageHeaderBase';
var propTypes = {
  /**
   * Icon node passed by PageHeader
   */
  icon: PropTypes.node,

  /**
   * Title node passed by PageHeader
   */
  title: PropTypes.node,

  /**
   * Info node passed by PageHeader
   */
  info: PropTypes.node,

  /**
   * Nav content which appears in the upper right hand corner.
   * 'navRight' prop will be deprecated soon, instaed use 'onRenderControls'
   */
  navRight: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
};

var Base = function Base(props) {
  return React.createElement("div", {
    className: "slds-grid slds-page-header__row"
  }, React.createElement("div", {
    className: "slds-col slds-page-header__col-title"
  }, React.createElement(MediaObject, {
    body: React.createElement("div", null, props.title, props.info),
    className: "slds-no-space slds-grow",
    figure: props.icon
  })), React.createElement("div", {
    className: "slds-col slds-no-flex slds-grid slds-align-top"
  }, props.onRenderControls ? props.onRenderControls : props.navRight));
};

Base.displayName = displayName;
Base.propTypes = propTypes;
export default Base;
